src/hooks/useAllCities.js

```diff
@@ -0,0 +1,51 @@
+import { useMemo } from 'react';
+import { useInfiniteQuery } from '@tanstack/react-query';
+import fetchCities from '../services/CitiesServise';
+import transformCityItem from '../data/transformCityItem';
+
+const ALL_CITIES_QUERY_KEY = 'allCities';
+
+const useAllCities = ({ pageSize, sortOrder }) => {
+  const {
+    data,
+    error,
+    isLoading,
+    fetchNextPage,
+    hasNextPage,
+    isFetchingNextPage,
+    ...result
+  } = useInfiniteQuery({
+    queryKey: [ALL_CITIES_QUERY_KEY],
+    queryFn: ({ pageParam = 1 }) =>
+      fetchCities({
+        page: pageParam,
+        pageSize: pageSize,
+        sortOrder: sortOrder,
+      }),
+    initialPageParam: 1,
+
```

Comment:
We can add staleTime param here because the list of all cities won't be updated so often

```suggestion
   staleTime: 1000 * 60 * 2,
```